### PR TITLE
Update go-connections package

### LINKED ISSUE #32: cannot bump go-connections version in docker due to compilation error
While trying to vendor go-connections in docker, as a follow up to #31, I get compilation errors. 

This is because of a change to 'NewUnixSocket', changing the type of group argument from string to int. 

https://github.com/docker/go-connections/blob/1b14b2d192e2f91cdc2bc6bf9aee0b0e116eed42/sockets/unix_socket.go#L11-L12

`// NewUnixSocket creates a unix socket with the specified path and group.
func NewUnixSocket(path string, gid int) (net.Listener, error) {
if err := syscall.Unlink(path); err != nil && !os.IsNotExist(err) {`

The NewUnixSocket interface now expects an integer as gid, instead of group name, which was the previous signature of 'NewUnixSocket'.

This breaks docker/docker, as the code depends on the version that accepted a string as group:

https://github.com/docker/docker/blob/40c51569e2a557e6656ea4314c824fb1eabb2c4e/pkg/listeners/listeners_unix.go#L34

`l, err := sockets.NewUnixSocket(addr, socketGroup)`

### LINKED ISSUE #30450: Docker 1.13 cli gives a warning about crypto/x509
(moving from https://github.com/docker/for-win/issues/422)

### Expected behavior
Docker ouput should be free of warnings (unless there's something to fix)

### Actual behavior
Docker 1.13.0 provides a developer warning on every run - it is reporting a library does not exist on windows. That's not something I can fix as a user.
```
>docker --version
time="2017-01-24T09:45:40-08:00" level=warning msg="Unable to use system certificate pool: crypto/x509: system root pool is not available on Windows"
Docker version 1.13.0, build 49bf474
```
### Information
  - host distribution and version (Windows version, build number, etc)
running on windows 10 pro, Version 1607 (OS Build 14292.693)

This might be an issue from Go's system cert pool: https://github.com/golang/go/issues/16736
This may also be fixed if https://github.com/docker/docker/issues/24806 is completed.

### Steps to reproduce the behavior

  1. Install docker 1.13.0 on windows
  2. Run `docker --version` (or any other command)

### PR DESCRIPTION
**Pull request by dmcgowan**
Changes windows message about system certificates from warn to info level. Ensures that server does not attempt to load an empty ca file.

fixes #30450
fixes https://github.com/docker/go-connections/issues/32

### CODE REVIEW

**Comment by cpuguy83** [comment]
`pkg/listeners/listeners_unix.go:34: cannot use socketGroup (type string) as type int in argument to sockets.NewUnixSocket`

**Comment by LK4D4** [comment]
ping @dmcgowan

**Comment by dmcgowan** [comment]
Updated to convert the socket group to an integer.  That seems like the intent of https://github.com/docker/go-connections/pull/26.

File: pkg/listeners/listeners_unix.go [hunk 1]
```diff
@@ -31,7 +31,12 @@ func Init(proto, addr, socketGroup string, tlsConfig *tls.Config) ([]net.Listene
 		}
 		ls = append(ls, l)
 	case "unix":
-		l, err := sockets.NewUnixSocket(addr, socketGroup)
+
+		gid, err := strconv.Atoi(socketGroup)
```
Comment (cpuguy83) [inline pkg/listeners/listeners_unix.go]: I think this really needs to be updated such that the function takes a gid as an integer, and then do group lookup in `cmd/dockerd/daemon.go`

**Comment by cpuguy83** [comment]
Pushed an update to this to use the proper group lookup and rebased.
Windows failure is some new thing that's failing everywhere (unrelated).

**Review by mlaventure — APPROVED** [review]
LGTM

ping @LK4D4

**Comment by tiborvass** [comment]
LGTM

**Comment by thaJeztah** [comment]
This was also in the 1.13 branch / 17.03 through https://github.com/docker/docker/pull/30730, so updated the milestone

**Pull request merged**